flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -173,6 +173,18 @@ public class TaskManagerOptions {
                                     + RPC_PORT.key()
                                     + "') will be used.");
 
+    /** The default port that <code>CollectSinkFunction$ServerThread</code> is using. */
+    @Documentation.Section({
+        Documentation.Sections.COMMON_HOST_PORT,
+        Documentation.Sections.ALL_TASK_MANAGER
+    })
+    public static final ConfigOption<Integer> COLLECT_PORT =
+            key("taskmanager.collect.port")
+                    .intType()
+                    .defaultValue(0)
+                    .withDescription(
+                            "The local port that the CollectSinkFunction$ServerThread binds to. The default is 0, which corresponds to a random port assignment.");
```

Comment:
I think the current description is kind of developer oriented, rather than user oriented. `CollectSinkFunction` is an `@Internal` class. Ideally, this should only be used for Flink's client, and users should not be aware of it. From a user's perspective, I think they only need to know that this port is for the client to retrieve query results from the TaskManager.